src/test/java/reactor/kafka/receiver/internals/CommitRetryTests.java

```diff
@@ -0,0 +1,128 @@
+        Set<TopicPartition> assigned = Collections.singleton(tp0);
+        CountDownLatch subscribeLatch = new CountDownLatch(1);
+        willAnswer(inv -> {
+            listener.set(inv.getArgument(1));
+            listener.get().onPartitionsAssigned(Collections.singletonList(tp0));
+            subscribeLatch.countDown();
+            return null;
+        }).given(consumer).subscribe(any(Collection.class), any(ConsumerRebalanceListener.class));
+        CountDownLatch commitLatch = new CountDownLatch(5);
+        willAnswer(inv -> {
+            OffsetCommitCallback callback = inv.getArgument(1);
+            callback.onComplete(inv.getArgument(0), new RetriableCommitFailedException("test"));
+            commitLatch.countDown();
+            return null;
+        }).given(consumer).commitAsync(any(), any());
+        final Map<TopicPartition, List<ConsumerRecord<Integer, String>>> records = new HashMap<>();
+        records.put(new TopicPartition("foo", 0), Arrays.asList(
+            new ConsumerRecord<>("foo", 0, 0L, 1, "foo"),
+            new ConsumerRecord<>("foo", 0, 1L, 1, "bar"),
+            new ConsumerRecord<>("foo", 0, 2L, 1, "baz"),
+            new ConsumerRecord<>("foo", 0, 3L, 1, "qux")));
+        ConsumerRecords<Integer, String> consumerRecords = new ConsumerRecords<>(records);
+        AtomicBoolean first = new AtomicBoolean(true);
+        willAnswer(inv -> {
+            Thread.sleep(10);
+            if (first.getAndSet(false)) {
+                return consumerRecords;
+            }
+            return ConsumerRecords.empty();
+        }).given(consumer).poll(any(Duration.class));
+        given(consumer.assignment()).willReturn(assigned);
+        ReceiverOptions<Object, Object> options = ReceiverOptions.create()
+            .commitInterval(Duration.ofSeconds(1))
+            .maxCommitAttempts(5)
+            .subscription(Collections.singletonList("foo"));
+        KafkaReceiver receiver = KafkaReceiver.create(cf, options);
+        List<ReceiverRecord<?, ?>> received = new ArrayList<>();
+        CountDownLatch receiveLatch = new CountDownLatch(4);
+        Disposable disposable = receiver.receive()
+            .publishOn(Schedulers.newSingle("retryCommits"))
+            .doOnNext(rec -> {
+                received.add((ReceiverRecord<?, ?>) rec);
+                receiveLatch.countDown();
+            })
+            .subscribe();
+        assertTrue(subscribeLatch.await(10, TimeUnit.SECONDS));
+        assertTrue(receiveLatch.await(10, TimeUnit.SECONDS));
+        received.get(0).receiverOffset().commit();
+        assertTrue(commitLatch.await(10, TimeUnit.SECONDS));
+        verify(consumer, times(5)).commitAsync(any(), any());
+        Map<TopicPartition, OffsetAndMetadata> commits = new HashMap<>();
+        commits.put(new TopicPartition("foo", 0), new OffsetAndMetadata(1L));
+        verify(consumer, times(5)).commitAsync(eq(commits), any());
+        disposable.dispose();
+    }
```

Comment:
would it be possible to add a test that validates the configured delay is used? maybe integrate that in the test above, since for now it only configures `commitInterval` and `maxCommitAttempts`?

Reply:
Good catch; thanks!